src/main/java/org/wildfly/security/password/impl/AbstractPasswordImpl.java

```diff
@@ -35,7 +35,9 @@
 
     abstract <S extends KeySpec> S getKeySpec(final Class<S> keySpecType) throws InvalidKeySpecException;
 
-    abstract boolean verify(final char[] guess) throws InvalidKeyException;
+    abstract boolean canVerify(Class<?> credentialType);
```

Comment:
Same comment.  This adds more complexity to callers since they have to heurstically guess what type of credential to submit.